File: Cpp_Parallel/src/lib/SerialSelectionSort.cpp
#include "include/SerialSelectionSort.h"

void SerialSelectionSort::sort(std::vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        int current_min = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[current_min]) {
                current_min = j;
            }
        }
        std::swap(arr[current_min], arr[i]);
    }
}

File: Java_Parallel/src/SerialSorts/SerialSelectionSort.java
package SerialSorts;

public class SerialSelectionSort {
    public static void sort(int[] a) {
        int n = a.length;
        for (int i = 0; i < n; i++) {
            int current_min = i;
            for (int j = i + 1; j < n; j++) {
                if (a[j] < a[current_min]) {
                    current_min = j;
                }
            }
            int temp = a[current_min];
            a[current_min] = a[i];
            a[i] = temp;
        }
    }
}
